python/openassetio_traitgen/generators/helpers.py

```diff
@@ -73,9 +73,23 @@ def package_dependencies(
     """
     dependencies = set()
     for declaration in declarations:
-        if isinstance(declaration, SpecificationDeclaration):
-            for trait in declaration.trait_set:
-                dependencies.add(trait.package)
+        dependencies |= set(declaration_dependencies(declaration))
+    dependencies = list(dependencies)
```

Comment:
[Consider] This loop is good, but there's an argument to be made that expressing this as a `reduce` conveys that you're trying to build a unique-set better, the `|=` accumulation in the loop reads a tad odd.

something like ...
```python
dependencies = reduce((lambda acc, dep: acc.union(set(declaration_dependencies(dep)))), declarations, set())
```

In either case though, a comment "Cycle through the declarations, and build a unique set of their dependencies" wouldn't go amiss.

Reply:
Refactored into one-liners in d62b361f. Hopefully a bit clearer (assuming you can wrap your head around double-for loops in Python list comprehensions).